lib/child_process.js

```diff
@@ -50,7 +50,20 @@ exports.fork = function(modulePath /*, args, options*/) {
 
   args = execArgv.concat([modulePath], args);
 
-  if (!Array.isArray(options.stdio)) {
+  if (typeof options.stdio === 'string') {
+    const stdioStringToArray = (option) => {
+      return [option, option, option, 'ipc'];
+    };
+    switch (options.stdio) {
+      case 'ignore':
+      case 'pipe':
+      case 'inherit':
+        options.stdio = stdioStringToArray(options.stdio);
+        break;
+      default:
+        throw new TypeError(`Incorrect type of stdio option: ${options.stdio}`);
```

Comment:
@sam-github I opted to go with the `TypeError: Unknown stdio option` although I do find it useful to have the corrupted input logged back to the developer. That being said,  I don't think it's useful enough to go against consistency unless we do decide that it is beneficial to change all errors accordingly. 

@addaleax I did have to use `util.inspect` to prevent printing `[Object Object]`. Not sure why I forgot to use that in the original PR